backend/chat-service/src/main/java/com/chatservice/services/ChatService.java

```diff
@@ -0,0 +1,110 @@
+package com.chatservice.services;
+
+import com.chatservice.communication.UserServiceCommunicationClient;
+import com.chatservice.dtos.*;
+import com.chatservice.entities.Conversation;
+import com.chatservice.entities.Message;
+import com.chatservice.mappers.ConversationMapper;
+import com.chatservice.mappers.GroupMemberMapper;
+import com.chatservice.repositories.ConversationRepository;
+import com.chatservice.repositories.GroupMemberRepository;
+import com.chatservice.repositories.MessageRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class ChatService {
+
+    private final ConversationRepository conversationRepository;
+
+    private final GroupMemberRepository groupMemberRepository;
+
+    private final ConversationMapper conversationMapper;
+
+    private final GroupMemberMapper groupMemberMapper;
+
+    private final UserServiceCommunicationClient userServiceCommunicationClient;
+
+    private final MessageRepository messageRepository;
+
+    public Conversation createPrivateConversation(NewPrivateConversationDTO newPrivateConversationDTO,
+                                                  UUID conversationCreatorUser){
+
+        if(newPrivateConversationDTO.getConversationMemberUser().equals(conversationCreatorUser)){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "UserId's must be different to create a conversation");
+        }
+
+        List<UUID> checkIfPrivateConversationExists = conversationRepository.findPrivateConversationForTwoUsers
+                (newPrivateConversationDTO.getConversationMemberUser(),
+                conversationCreatorUser);
+
+        if(!checkIfPrivateConversationExists.isEmpty()){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+                    "Conversation with user: " + newPrivateConversationDTO.getConversationMemberUser() + " already exists");
+        }
+
+        Conversation createdConversation = conversationRepository.save(conversationMapper.toEntity(false));
```

Comment:
Tutaj mapper do ustawienia jednej hardcodowanej zmiennej bym odpuścił bo ciężko się to czyta.